server/v2/store/commands.go

```diff
@@ -0,0 +1,155 @@
+		Args:    cobra.RangeArgs(0, 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// bind flags to the Context's Viper so we can get pruning options.
+			vp := viper.New()
+			if err := vp.BindPFlags(cmd.Flags()); err != nil {
+				return err
+			}
+			if err := vp.BindPFlags(cmd.PersistentFlags()); err != nil {
+				return err
+			}
+
+			pruningOptions, err := getPruningOptionsFromCmd(cmd, args)
+			if err != nil {
+				return err
+			}
+
+			cmd.Printf("get pruning options from command flags, strategy: %v, keep-recent: %v\n",
+				pruningOptions.Strategy,
+				pruningOptions.KeepRecent,
+			)
+
+			logger := log.NewLogger(cmd.OutOrStdout())
+			app := appCreator(logger, vp)
+			store := app.GetStore()
+
+			rootStore, ok := store.(storev2.RootStore)
+			if !ok {
+				return fmt.Errorf("currently only support the pruning of rootmulti.Store type")
+			}
+			latestHeight, err := rootStore.GetLatestVersion()
+			if err != nil {
+				return err
+			}
+
+			// valid heights should be greater than 0.
+			if latestHeight <= 0 {
+				return fmt.Errorf("the database has no valid heights to prune, the latest height: %v", latestHeight)
+			}
+
+			pruningHeight := latestHeight - pruningOptions.KeepRecent
+			cmd.Printf("pruning heights up to %v\n", pruningHeight)
+
+			err = rootStore.Prune(pruningHeight)
+			if err != nil {
+				return err
+			}
+
+			cmd.Println("successfully pruned the application root multi stores")
+			return nil
+		},
+	}
+
+	cmd.Flags().String(FlagAppDBBackend, "", "The type of database for application and snapshots databases")
+	cmd.Flags().Uint64(FlagPruningKeepRecent, 0, "Number of recent heights to keep on disk (ignored if pruning is not 'custom')")
+	cmd.Flags().Uint64(FlagPruningInterval, 10,
+		`Height interval at which pruned heights are removed from disk (ignored if pruning is not 'custom'), 
+		this is not used by this command but kept for compatibility with the complete pruning options`)
+
+	return cmd
+}
+
+func getPruningOptionsFromCmd(cmd *cobra.Command, args []string) (pruningtypes.PruningOptions, error) {
+	rootViper := serverv2.GetViperFromCmd(cmd)
+
+	var pruning string
+	if len(args) > 0 {
+		// use the first argument if present to set the pruning method
+		pruning = args[0]
+	} else {
+		// If we not pass pruning option to the first argument,
+		// Get it from app.toml
+		// If we not do any modify to app.toml, should be default prune in there.
+		pruning = rootViper.GetString(FlagPruning)
+	}
+
+	strategy := strings.ToLower(pruning)
+
+	switch strategy {
+	case pruningtypes.PruningOptionDefault, pruningtypes.PruningOptionNothing, pruningtypes.PruningOptionEverything:
+		return pruningtypes.NewPruningOptionsFromString(strategy), nil
+
+	case pruningtypes.PruningOptionCustom:
+		var (
+			pruningKeepRecent uint64
+			pruningInterval   uint64
+			err               error
+		)
+
+		if cmd.Flags().Changed(FlagPruningKeepRecent) {
```

Comment:
Prioritize reading from flag first. If flag is not used, then read from `app.toml`